### DISCUSSION TITLE
What happended to me ?

### DISCUSSION THREAD

**Discussion by sky124380729** [vuejs/vitepress#3018] [Q&A] [answered]
### Describe the bug

when i created vitepress in a seperate project, it was ok.
but when i created vitepress in a monorepo, it told error..
```bash
ENOENT: no such file or directory, realpath 'E:\monorepos\equipment-product\node_modules\.pnpm\vitepress@1.0.0-rc.20_@algolia+client-search@4.20.0_@types+node@18.17.18_axios@1.5.0_postcss@_cearrku5wuarlambwx5odjeta4\node_modules\vitepress\dist\client\theme-default\components\VPLocalNavOutlineDropdown.vue'
```

### Reproduction

![image](https://github.com/vuejs/vitepress/assets/23203328/affbc05d-dc66-4283-8b01-f875dfda8b5f)

### Expected behavior

i don't know

### System Info

```Text
System:
    OS: Windows 10 10.0.17763
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 3.62 GB / 15.41 GB
  Binaries:
    Node: 18.5.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Local\pnpm\yarn.CMD
    npm: 8.12.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.7.0 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Internet Explorer: 11.0.17763.2145
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Can you share the repo?

Firstly, delete lock file from docs/ directory. Second, make sure docs is there in pnpm-workspaces. Then run `pnpm install`

Next, if you've vitepress listed as a dep of `docs/package.json`, then you need to add these scripts there:

```json
{
  "name": "docs",
  "private": true,
  "type": "module",
  "scripts": {
    "dev": "vitepress dev",
    "build": "vitepress build",
    "preview": "vitepress preview"
  },
  "devDependencies": {
    "vitepress": "latest"
  }
}
```

Then in your root `package.json`, you need to have these scripts:

```json
{
  "scripts": {
    "docs:dev": "pnpm -F=docs dev",
    "docs:build": "pnpm -F=docs build",
    "docs:preview": "pnpm -F=docs preview"
  }
}
```

If you don't have vitepress as a dependency in your `docs/package.json`, then simply delete it. Remove `docs` from workspaces. Install vitepress in root. Keep using `vitepress dev docs`.

**Comment by sky124380729** [comment]
Due to company regulations，i can not share my repo...
![image](https://github.com/vuejs/vitepress/assets/23203328/3eb10ecf-a44a-4943-9871-4b02c3471176)
![image](https://github.com/vuejs/vitepress/assets/23203328/5c2a6c1f-e6e6-439b-975f-c3b8ce015b71)

i found that sometimes is works fine, but when I modify other project's source code. it will fail. I haven't found the cause of the problem yet.
it's realy strange...i haven't found any code could effect vitepress,maybe vite.config.ts in root?

**Comment by sky124380729** [comment, marked as answer]
i update my node version then the problem resolved
